File: get_pins.py
import requests
from bs4 import BeautifulSoup
import sys

urls = sys.argv[1:]
for url in urls:
    try:
        r = requests.get(url, headers={'User-Agent': 'Mozilla/5.0'}, timeout=10)
        soup = BeautifulSoup(r.text, 'html.parser')
        og_img = soup.find('meta', property='og:image')
        if og_img:
            print(f"{url} -> {og_img['content']}")
        else:
            print(f"{url} -> Not found")
    except Exception as e:
        print(f"{url} -> Error: {e}")

File: get_pins.js
const https = require('https');

const urls = process.argv.slice(2);

function fetchUrl(url) {
  return new Promise((resolve, reject) => {
    https.get(url, (res) => {
      if (res.statusCode >= 300 && res.statusCode < 400 && res.headers.location) {
        let location = res.headers.location;
        if (!location.startsWith('http')) {
          location = new URL(location, url).href;
        }
        return resolve(fetchUrl(location));
      }
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        const match = data.match(/<meta\s+property="og:image"\s+content="([^"]+)"/i);
        if (match) {
          resolve(match[1]);
        } else {
          resolve('Not found');
        }
      });
    }).on('error', reject);
  });
}

async function run() {
  for (const url of urls) {
    try {
      const img = await fetchUrl(url);
      console.log(`${url} -> ${img}`);
    } catch (e) {
      console.log(`${url} -> Error: ${e.message}`);
    }
  }
}
run();
